operator/controllers/btpoperator_controller_test.go

```diff
@@ -152,34 +177,52 @@ var _ = Describe("BTP Operator controller", func() {
 		})
 
 		It("soft delete (after timeout) should succeed", func() {
-			reconciler.SetTimeout(testTimeout)
-			reconciler.Client = newFakeK8s(reconciler.Client)
-
-			triggerDelete()
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateDeleting))
+			Eventually(isCrNotFound).WithTimeout(deleteTimeout).WithPolling(crStatePollingIntevral).Should(BeTrue())
 			doChecks()
 		})
 
 		It("soft delete (after hard deletion fail) should succeed", func() {
-			reconciler.SetTimeout(time.Minute * 1)
 			reconciler.Client = newFakeK8s(reconciler.Client)
 
-			triggerDelete()
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateDeleting))
+			Eventually(isCrNotFound).WithTimeout(deleteTimeout).WithPolling(crStatePollingIntevral).Should(BeTrue())
 			doChecks()
 		})
 
 		It("hard delete should succeed", func() {
-			reconciler.SetTimeout(time.Minute * 1)
-
+			Expect(k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: btpOperatorName}, cr)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, cr)).To(Succeed())
+			Eventually(getCurrentCrState).WithTimeout(stateChangeTimeout).WithPolling(crStatePollingIntevral).Should(Equal(types.StateDeleting))
+			Eventually(isCrNotFound).WithTimeout(deleteTimeout).WithPolling(crStatePollingIntevral).Should(BeTrue())
 			doChecks()
 		})
 	})
```

Comment:
All three test test same flow which is happening in the cluster. We need to fix them.

Reply:
fixed